packages/devtools_app/lib/src/code_size/code_size_attribution.dart

```diff
@@ -0,0 +1,310 @@
+              node.display,
+              overflow: TextOverflow.ellipsis,
+            ),
+          ),
+        ),
+        const Padding(
+          padding: EdgeInsets.symmetric(horizontal: densePadding),
+          child: Icon(Icons.arrow_forward),
+        ),
+      ],
+    );
+  }
+
+  void _selectMainNode(CallGraphNode node) {
+    setState(() {
+      this.node = node;
+    });
+  }
+}
+
+class FromColumn extends ColumnData<CallGraphNode> {
+  FromColumn() : super('From');
+
+  @override
+  String getValue(CallGraphNode dataObject) => dataObject.display;
+}
+
+class ToColumn extends ColumnData<CallGraphNode> {
+  ToColumn() : super('To');
+
+  @override
+  ColumnAlignment get alignment => ColumnAlignment.right;
+
+  @override
+  String getValue(CallGraphNode dataObject) => dataObject.display;
+}
+
+class DominatorTree extends StatelessWidget {
+  DominatorTree({
+    @required this.dominatorTreeRoot,
+    @required this.selectedNode,
+  });
+
+  static const dominatorTreeTableKey = Key('DominatorTree - table');
+
+  final DominatorTreeNode dominatorTreeRoot;
+
+  final CallGraphNode selectedNode;
+
+  final _packageColumn = _PackageColumn();
+
+  @override
+  Widget build(BuildContext context) {
+    _expandToSelected();
+    // TODO(kenz): programmatically select [selectedNode] in the table.
+    return TreeTable<DominatorTreeNode>(
+      key: dominatorTreeTableKey,
+      dataRoots: [dominatorTreeRoot],
+      columns: [_packageColumn],
+      treeColumn: _packageColumn,
+      keyFactory: (node) => PageStorageKey<String>('${node.callGraphNode.id}'),
+      sortColumn: _packageColumn,
+      sortDirection: SortDirection.descending,
+      autoExpandRoots: true,
+    );
+  }
+
+  void _expandToSelected() {
+    var selected = dominatorTreeRoot.firstChildWithCondition(
+        (node) => node.callGraphNode.id == selectedNode.id);
+
+    while (selected != null) {
+      selected.expand();
+      selected = selected.parent;
+    }
+  }
+}
+
+class _PackageColumn extends TreeColumnData<DominatorTreeNode> {
+  _PackageColumn() : super('Package');
+
+  @override
+  String getValue(DominatorTreeNode dataObject) =>
+      dataObject.callGraphNode.display;
+
+//  @override
+//  bool get supportsSorting => false;
+}
+
+extension CallGraphNodeDisplay on CallGraphNode {
+  String get display {
+    final displayText =
+        data is ProgramInfoNode ? data.qualifiedName : data.toString();
+    if (displayText == '@shared') {
+      // Special case '@shared' because this is the name of the call graph root,
+      // and '@root' has a more intuitive meaning.
+      return '@root';
+    }
+    return displayText;
+  }
+
+  CallGraphNode get dominatorRoot {
+    var root = this;
+    while (root.dominator != null) {
+      root = root.dominator;
+    }
+    return root;
+  }
+}
+
+class DominatorTreeNode extends TreeNode<DominatorTreeNode> {
+  DominatorTreeNode._(this.callGraphNode);
+
+  static DominatorTreeNode from(CallGraphNode cgNode) {
```

Comment:
general style nit. why a static method instead of a factory constructor?

Reply:
changed to factory